### ISSUE TITLE
PeerServer Cloud service down?

### ISSUE THREAD

**Issue by fcor** [peers/peerjs#939]
Is the PeerServer Cloud service down?

I can't establish any connection on some existing prototypes that were working fine a couple of days ago.

This is what I see on firefox console when trying to use the demo from the peer.js website:

![Capture d’écran 2022-03-25 à 11 21 27](https://user-images.githubusercontent.com/21111451/160102762-74a4c4e1-9697-4153-aa0a-6d1c01e2997a.png)

I would be happy to help if something is broken and needs some refactor :)

**Comment by julienreszka** [comment]
See https://github.com/peers/peerjs/issues/937#issuecomment-1077712367

**Comment by kidandcat** [comment]
Yes, it is down, we can no longer work on the PeerJS project and we are looking for someone to take care of it: https://github.com/peers/peerjs/issues/938

**Comment by MajorDeanSimba** [comment]
We created a Video chat service for our 2.5Million members based on your promise that Peerjs would be working!
You can see it here: https://www.anoox.com/chat/video_chat_description.php
What are we to do now?
How could you just discontinue something that We the People were counting on without notice :(

FYI, I have looked at your instructions "run your own PeerServer" at here: https://github.com/peers/peerjs-server
but it is not working!

This is very urgent: you may contact me directly at dean_ansari@anoox.com

**Comment by coderGtm** [comment]
Even I have a [game](https://play.google.com/store/apps/details?id=com.coderGtm.microchess)  live on Play Store which uses your service for multiplayer gaming. Now except the v/s Computer mode, the game is rendered useless. And I think that deploying the server on Heroku does not permit commercial use in free plan. Now what shall we do?? :-(

**Comment by kidandcat** [comment]
Well, you NEED a server to use WebRTC, and this is a community project. We make our best effort to keep it online, but this is not a service offered by a company and there is no guarantee of SLA.

You have the facilities to setup your own server in one click:
https://github.com/peers/peerjs/issues/937#issuecomment-1077712367

And finally, if you have production apps running relying on PeerJS, you should have your own PeerServer so your service dependes entirely of you.

@MajorDeanSimba You should be able to habe your own PeerServer running in minutes following these instructions:
https://github.com/peers/peerjs/issues/937#issuecomment-1077712367

@coderGtm If you have comercial production games relying on PeerJS, you should consider paying your own server. Be it on heroku, or any other provider.

**Comment by coderGtm** [comment]
I don't make any money as of now from my games so paying for a server seems far-fetched. Any other recommendations?? 
Thanks:-)

**Comment by kidandcat** [comment]
If you don't make any money, you shouldn't have any issues with commercial limits, as you are not commercialising anything

**Comment by MajorDeanSimba** [comment]
Hello KidandCat,

So we would welcome installing the PeerJs server on our own server and generating the Peer IDs on our own.
We really should have done this at the start but now this shutdown of the Peerjs cloud server is of course time to do so.
FYI, our server is a dedicated LAMP server, running on CentOS 7.
So my question is how do we do this?  I see in your reply you have stated to do this:

const myPeer = new Peer( {
     secure:true,
     host: 'my-peerjs-server-name.herokuapp.com', // change here the herokuapp name
     port: 443
})

Questions: 
1- This is to go into the server side index.js I assume?
2- what should be the value of  host: 'my-peerjs-server-name.herokuapp.com' ?

Again we want to run this on our server and not outside such as heroku

And after we got it working on server, then how do we point the client side, the browser, to get load peerjs.min.js from our server to generate the Peer ID from this local resource?

Thanks

P.S., I would be willing to pay you to get on Zoom or anyDesk with me to take care off this quick since our
Video chat service has been done for days now due to this failure of Peerjs not loading.

**Comment by MajorDeanSimba** [comment]
Hey KidandCat,

So this is exactly what we have done to install and run PeerJs server on our server and to get the PeerId on client side from it:

1st, We are now getting the Peerjs code locally via:
<script src="peerjs_source.js"></script>

2nd, on server side we installed peer:  npm install peer
added to the server side index.js then:
const { PeerServer } = require('peer');
const peerServer = PeerServer({ port: 4000, path: '/myapp' });

node.js Server starts OK, indicating that:  listening on *:3000

on client side then we added this:

myPeer = new Peer(49, {
      host: 'localhost',
      port: 4000,
      path: '/myapp'
    });

However it fails with this Error message:
WebSocket connection to 'wss://localhost:4000/myapp/peerjs?key=peerjs&id=49&token=orkbd8usbkh' failed: 

And FYI:
1- On server we are running Centos 7
2- for SSL we use LetsEncrypt
3- Our server is Apache

Thank you for your prompt reply since this is very URGENT as our members can no longer use the Video meeting
service which was getting the Peer ID for connecting Videos via your Peerjs 

Regards,
Dean @ Anoox.com

**Comment by Rainb0wCodes** [comment]
@MajorDeanSimba Are you sure you can use TLS on localhost?

**Comment by MajorDeanSimba** [comment]
Hello Rainbow,

Not sure about that question. And I Googled and could not see a sensible answer.
So how do I check for the answer to your question?

BTW the test (development) version of this Video chat page is here:

https://www.anoox.com/chat/video_chat_txt.php?idv=2492160&kst=3082420

Note: the member you are (Video) calling needs to be logged in to answer your call.
And of course right now where they to answer it there would be no video meeting due to this Peerjs thing.
But in case you can get your answer to your question via above page I have included it.

Thanks as this problem is very Urgent now

**Comment by PaulFidika** [comment]
Are there any WebRTC libraries anyone would recommend to replace PeerJS with? I like it, I'm just wondering what all is out there and what people would recommend.

**Comment by jonasgloning** [comment]
Should be up again 🎉
